util/predicates/secret_predicates.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package predicates
+
+import (
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// SecretCreateUpdate returns a predicate that returns true for a Secret create event
+func SecretCreateUpdate(logger logr.Logger) predicate.Funcs {
+	log := logger.WithValues("predicate", "SecretCreateOrUpdate")
+
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			log = log.WithValues("eventType", "create")
+
+			c, ok := e.Object.(*corev1.Secret)
+			if !ok {
+				log.V(4).Info("Expected Secret", "type", e.Object.GetObjectKind().GroupVersionKind().String())
+				return false
+			}
+			log = log.WithValues("namespace", c.Namespace, "secret", c.Name)
+			log.V(4).Info("Expected Secret")
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			log = log.WithValues("eventType", "update")
+
+			_, okOld := e.ObjectOld.(*corev1.Secret)
+			newSecret, okNew := e.ObjectNew.(*corev1.Secret)
+			if !okOld || !okNew {
+				log.V(4).Info("Expected Secret", "type", e.ObjectOld.GetObjectKind().GroupVersionKind().String())
+				return false
+			}
+			log = log.WithValues("namespace", newSecret.Namespace, "secret", newSecret.Name)
+
+			return true
+		},
```

Comment:
Should we filter out these secrets by `type`, so we only enqueue for requests that are relevant for our use case?